lib/gtl_formatter.rb

```diff
@@ -104,7 +107,12 @@ def self.format_cols(view, row, controller, options)
         celltext, span = send(special_cases[view.extras[:filename]][view.col_order[col_idx]], row[col])
       elsif COLUMN_WITH_IMAGE.key?(col)
         # Generate html for the list icon
-        icon, icon2, image = send(COLUMN_WITH_IMAGE[col], record)
+        # NOTE: COLUMN_WITH_IMAGE basically whitelists these associations
+        # determine the record that needs to be decorated (follow the periods)
+        associations = col.split('.')[..-2]
+        col_record = associations.inject(record) { |r, association_name| r.send(association_name) }
+        # NOTE: currently this only calls fonticon_or_fileicon
+        icon, icon2, image = send(COLUMN_WITH_IMAGE[col], col_record)
```

Comment:
so we used to send the row record to get the icons.
Now we are going to send the record in question to choose the icons.

This fails for resource_pool because it has special cases for determining the icons. This works for vm because it has the same logic as ems.

before
======

the icon for the `ems.name` was `resource_pool.decorator.font_icon` (the `record` passed in was the `resource_pool`)

after
=====

the icon for `ems.name` is `ems.decorator.font_icon` (the `record` passed in is now `resource_pool.ext_management_system` via the `inject`)